client/dashboard/task-list/tasks/shipping/rates.js

```diff
@@ -0,0 +1,225 @@
+							data: {
+								enabled: false,
+							},
+						} );
+					} );
+				}
+				return;
+			}
+
+			if ( flatRateMethods.length ) {
+				// Update the existing method.
+				apiFetch( {
+					method: 'POST',
+					path: `/wc/v3/shipping/zones/${ zone.id }/methods/${ flatRateMethods[ 0 ].instance_id }`,
+					data: {
+						enabled: true,
+						settings: { cost: zoneRates[ zone.id ] },
+					},
+				} );
+			} else {
+				// Add new method if one doesn't exist.
+				apiFetch( {
+					method: 'POST',
+					path: `/wc/v3/shipping/zones/${ zone.id }/methods`,
+					data: {
+						method_id: 'flat_rate',
+						settings: { cost: zoneRates[ zone.id ] },
+					},
+				} );
+			}
+		} );
+
+		// @todo This is a workaround to force the task to mark as complete.
+		// This should probably be updated to use wc-api so we can fetch shipping methods.
+		wcSettings.onboarding.shippingZonesCount = 1;
+
+		createNotice( 'success', __( 'Your shipping rates have been updated.', 'woocommerce-admin' ) );
+
+		this.props.completeStep();
+	}
+
+	handleChange( zoneId, value ) {
+		this.setState( prevState => ( { zoneRates: { ...prevState.zoneRates, [ zoneId ]: value } } ) );
+	}
+
+	handleToggle( zoneId ) {
+		this.setState( prevState => ( { enabledZones: xor( prevState.enabledZones, [ zoneId ] ) } ) );
+	}
+
+	handleBlur( zoneId ) {
+		const { zoneRates } = this.state;
+		zoneRates[ zoneId ] = getCurrencyFormatString( zoneRates[ zoneId ] );
+
+		this.setState( { zoneRates } );
+	}
+
+	render() {
+		const { enabledZones, zoneRates } = this.state;
+		const { shippingZones } = this.props;
+
+		if ( ! shippingZones.length ) {
+			return null;
+		}
+
+		return (
+			<Fragment>
+				<div className="woocommerce-shipping-rates">
+					{ shippingZones.map( zone => (
+						<div className="woocommerce-shipping-rate" key={ zone.id }>
+							<div className="woocommerce-shipping-rate__icon">
+								{ zone.locations ? (
+									zone.locations.map( location => (
+										<Flag size={ 24 } code={ location.code } key={ location.code } />
+									) )
+								) : (
+									// Icon used for zones without locations or "Rest of the world".
+									<i className="material-icons-outlined">public</i>
+								) }
+							</div>
+							<div className="woocommerce-shipping-rate__main">
+								<div className="woocommerce-shipping-rate__name">
+									{ zone.name }
+									{ zone.toggleEnabled && (
+										<FormToggle
+											checked={ enabledZones.includes( zone.id ) }
+											onChange={ () => this.handleToggle( zone.id ) }
+										/>
+									) }
+								</div>
+								{ ( ! zone.toggleEnabled || enabledZones.includes( zone.id ) ) && (
+									<div
+										className={ classnames( 'woocommerce-shipping-rate__control-wrapper', {
+											'has-value': zoneRates[ zone.id ],
+										} ) }
+									>
+										<span className="woocommerce-shipping-rate__control-prefix">
```

Comment:
This can be addressed in a follow-up issue/PR, however not all currencies will have the symbol as a prefix.

Maybe we could lift https://github.com/woocommerce/woocommerce-admin/blob/0f20052de6f9b706baaccfc6f8cb64c713c53d37/packages/components/src/filters/advanced/number-filter.js#L72-L96 into an exportable function or maybe a `PriceInput` component that does the handling for us (prior art: https://github.com/Automattic/wp-calypso/blob/97569a3c821c111dba318498cdf79d45db32a217/client/extensions/woocommerce/components/price-input/index.js).

Reply:
Added a temporary fix for now.  I agree a `PriceInput` component would be a good way to go about this.  I'm thinking we should hold on until we move away from newspack.